airflow/providers/amazon/aws/transfers/snowflake_to_s3.py

```diff
@@ -0,0 +1,150 @@
+        'externalbrowser' to authenticate using your web browser and
+        Okta, ADFS or any other SAML 2.0-compliant identify provider
+        (IdP) that has been defined for your account
+        'https://<your_okta_account_name>.okta.com' to authenticate
+        through native Okta.
+    :type authenticator: str
+    :param session_parameters: You can set session-level parameters at
+        the time you connect to Snowflake
+    :type session_parameters: dict
+    """
+
+    template_fields = (
+        's3_key',
+        's3_bucket',
+        'table_or_query',
+        'snowflake_conn_id',
+    )
+    template_ext = ('.sql',)
+    ui_color = '#ffebb2'
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        stage: Optional[str] = None,
+        s3_bucket: Optional[str],
+        s3_key: str,
+        query_or_table: str,
+        file_format: Optional[str],
+        database: Optional[str] = None,
+        warehouse: Optional[str] = None,
+        snowflake_conn_id: str = "snowflake_default",
+        unload_options: Optional[list] = None,
+        autocommit: bool = True,
+        include_header: bool = False,
+        role: Optional[str] = None,
+        authenticator: Optional[str] = None,
+        session_parameters: Optional[dict] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.stage = stage
+        self.s3_bucket = s3_bucket
+        self.s3_key = s3_key
+        self.file_format = file_format
+        self.warehouse = warehouse
+        self.database = database
+        self.snowflake_conn_id = snowflake_conn_id
+        self.query_or_table = query_or_table
+        self.unload_options = unload_options or []  # type: List
+        self.autocommit = autocommit
+        self.include_header = include_header
+        self.role = role
+        self.authenticator = authenticator
+        self.session_parameters = session_parameters
+
+    def execute(self, context):
+        snowflake_hook = SnowflakeHook(
+            snowflake_conn_id=self.snowflake_conn_id,
+            warehouse=self.warehouse,
+            database=self.database,
+            role=self.role,
+            authenticator=self.authenticator,
+            session_parameters=self.session_parameters,
+        )
+        unload_options = '\n\t'.join(self.unload_options)
+
+        if self.stage:
+            unload_query = f"""
+            COPY INTO @{self.stage}/{self.s3_key}
+            FROM ({self.query_or_table})
+            """
+
+            if self.file_format:
+                unload_query += f'FILE_FORMAT = ({self.file_format})'
+
+        else:
+            unload_query = f"""
+            COPY INTO 's3://{self.s3_bucket}/{self.s3_key}'
+            FROM ({self.query_or_table})
+            STORAGE_INTEGRATION = S3
+            FILE_FORMAT = ({self.file_format})
+            """
+
+        unload_query += f"""{unload_options}
+                            HEADER = {self.include_header};"""
```

Comment:
There's a lot of leading white spaces in all strings - is this intended? I'm wondering how this will render in log

Reply:
@turbaszek  i previously handled this with an `unindent_auto` function just for sql... if you think this would be helpful in the repo please suggest a location and i'll make a PR